lib/propshaft/helper.rb

```diff
@@ -6,20 +6,24 @@ def compute_asset_path(path, options = {})
 
     # Add an option to call `stylesheet_link_tag` with `:all` to include every css file found on the load path.
     def stylesheet_link_tag(*sources, **options)
-      if sources.first == :all
-        super(*all_stylesheets_paths, **options)
+      case sources.first
+      when :all
+        super(*all_stylesheets_paths , **options)
+      when :app
+        super(*app_stylesheets_paths , **options)
       else
         super
       end
     end
 
     # Returns a sorted and unique array of logical paths for all stylesheets in the load path.
     def all_stylesheets_paths
-      Rails.application.assets.load_path
-        .assets(content_types: [ Mime::EXTENSION_LOOKUP["css"] ])
-        .collect { |css| css.logical_path.to_s }
-        .sort
-        .uniq
+      Rails.application.assets.load_path.asset_paths_by_type("css")
+    end
+
+    # Returns a sorted and unique array of logical paths for all stylesheets in app/assets/**/*.css.
+    def app_stylesheets_paths
+      Rails.application.assets.load_path.asset_paths_by_glob("#{Rails.root.join("app/assets")}/**/*.css")
```

Comment:
Since this is only intended for loading styles, is there any reason to look in folders other than just stylesheets?

Reply:
This was needed to get a single-line matcher for fnmatch that works with both nested directories and not. Don't think it really matters, and it does provide slightly more flexibility if someone organizes their app/assets differently.